### ISSUE TITLE
Add an empty `NetworkParameters` struct to `Network::Testnet`

### ISSUE THREAD

**Issue by arya2** [ZcashFoundation/zebra#7968]
## Motivation 

Zebra currently hard-codes some consensus parameters that may need to be changed for testing.

Depends-On: #8325, #8326

### Possible Design

- Add an empty and optional `TestnetParameters` struct in `Network::Testnet`
- Add a `Network::new_testnet()` method that returns `Network::Testnet { ..default_testnet_params }`
  - Mass replace any remaining instances of `Testnet` in tests with `Network::new_testnet()`

Example (with fields that shouldn't be added yet):
 ```rust
Testnet(Arc<NetworkParameters { 
    cache_name,
    genesis_hash,
    network_id,
    default_port,
    activation_heights
>)
```

**Comment by AloeareV** [comment]
>     1. refactor tests to use `is_mainnet()` to look up test vectors using new functions in `zebra-test`, rather than matching on `Testnet`. This should be quick and easy to approve since it's only test code, and there are only mainnet and default public testnet test vectors. (No custom testnet vectors.)

If I'm going to implement this, is the intent here to replace the `match network` blocks with `if network.is_mainnet()` blocks? New functions in zebra-test like some kind of `fn test_vector_for_network` that gets offloaded to?

**Comment by arya2** [comment]
> If I'm going to implement this, is the intent here to replace the `match network` blocks with `if network.is_mainnet()` blocks?

We want to do this to minimize the changes in the following PRs where the `Testnet` variant is modified, but this issue is for adding an empty struct in the `Testnet` variant of `Network`. Replacing `match network` blocks with `if network.is_mainnet()` in tests should precede this issue as an early step in #7845, but we're still planning #7845 so there isn't an open issue for this yet.

We decided PR #7924 was too large to test or review accurately, and because this is a high-risk area of the code, we want to divide the changes across multiple PRs, where each PR adds one configurable parameter and `Network` method with unit tests for its changes.

We also discussed #1135 around the time that #7924 was opened, and decided to add any configurable parameters to the `Testnet` variant instead of adding a new variant or refactoring `Network` into a trait to reduce scope.
